cmd/list/accessrequests/cmd.go

```diff
@@ -0,0 +1,116 @@
+package accessrequests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	v1 "github.com/openshift-online/ocm-sdk-go/accesstransparency/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+const (
+	use   = "access-request"
+	short = "List access requests"
+	long  = "List access requests in Pending or Approved status. " +
+		"If '--cluster' flag is used, list all access requests in any status for the specified cluster."
+	example = `  # List all access requests for cluster 'foo'
+  rosa list access-request --cluster foo
+  `
+)
+
+func NewListAccessRequestsCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: []string{"accessrequest", "accessrequests", "access-requests"},
+		Short:   short,
+		Long:    long,
+		Example: example,
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), ListAccessRequestsRunner()),
+		Args:    cobra.NoArgs,
+	}
+
+	output.AddFlag(cmd)
+	ocm.AddOptionalClusterFlag(cmd)
+	return cmd
+}
+
+func ListAccessRequestsRunner() rosa.CommandRunner {
+	return func(_ context.Context, r *rosa.Runtime, cmd *cobra.Command, _ []string) error {
+		clusterId := ""
+		if cmd.Flags().Changed("cluster") {
+			cluster, err := r.OCMClient.GetCluster(r.GetClusterKey(), r.Creator)
+			if err != nil {
+				return err
+			}
+			clusterId = cluster.ID()
+		}
+		accessRequests, err := r.OCMClient.ListAccessRequest(clusterId)
+		if err != nil {
+			return err
+		}
+		if output.HasFlag() {
+			output.Print(accessRequests)
+		} else {
+			if len(accessRequests) == 0 {
+				if clusterId == "" {
+					r.Reporter.Infof("There are no Access Requests in Pending or Approved status.")
+				} else {
+					r.Reporter.Infof("There are no Access Requests for cluster '%s'.", r.ClusterKey)
+				}
+				return nil
+			}
+			writer := tabwriter.NewWriter(os.Stdout, 0, 4, 2, ' ', 0)
+			output, hasPending, pendingId := getAccessRequestsOutput(clusterId, accessRequests)
+			fmt.Fprint(writer, output)
+			if err := writer.Flush(); err != nil {
+				return err
+			}
+			if hasPending {
+				r.Reporter.Infof("Run the following command to approve or deny the access request:\n\n"+
+					"   rosa create decision --access-request %s --decision Approved\n"+
+					"   rosa create decision --access-request %s --decision Denied --justification \"justification\"\n",
+					pendingId, pendingId)
+			}
+		}
+		return nil
+	}
+}
+
+func getAccessRequestsOutput(clusterId string, accessRequests []*v1.AccessRequest) (string, bool, string) {
+	output := "STATE\tID\tCLUSTER ID\tUPDATED AT\n"
+	if clusterId != "" {
```

Comment:
@marcolan018 I'm curious...is `clusterId` ever an empty string? IIRC, when we create the AccessRequest we lookup the `Subscription` and fetch the `clusterId` from there 🤔

Reply:
@marcolan018 Ah I see the flow here. You only print out the `clusterId` if the user has passed it as a CLI parameter.

_Personally_, I would always print out the `clusterId` when showing details of the Access Request as it's a key part of the information required by customers.

Reply:
@robpblake 
will it be redundant to display the `clusterId` if user already specified the `--cluster` flag? Also in the whole list the value will be the same.

Reply:
hmmm, since user maybe set cluster name as the identity, I changed to always include clusterId in the output